File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char, char> sToT;
        unordered_map<char, char> tToS;

        for (int i = 0; i < s.size(); i++) {

            if (sToT.count(s[i])) {
                if (sToT[s[i]] != t[i])
                    return false;
            } else {
                sToT[s[i]] = t[i];
            }

            if (tToS.count(t[i])) {
                if (tToS[t[i]] != s[i])
                    return false;
            } else {
                tToS[t[i]] = s[i];
            }
        }

        return true;
      
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        sf={}
        tf={}
        if len(s)!=len(t):
            return False
        for sch, tch in zip(s, t):
            if sch in sf and  sf[sch]!=tch:
                return False
            if tch in tf and tf[tch]!=sch:
                return False
            sf[sch] = tch
            tf[tch] = sch
        return True